electrum/lnrouter.py

```diff
@@ -554,14 +566,6 @@ def get_distances(
                     continue
                 edge_startnode = channel_info.node2_id if channel_info.node1_id == edge_endnode else channel_info.node1_id
                 is_mine = edge_channel_id in my_channels
-                if is_mine:
-                    if edge_startnode == nodeA:  # payment outgoing, on our channel
-                        if not my_channels[edge_channel_id].can_pay(amount_msat, check_frozen=True):
-                            continue
```

Comment:
Are you sure it is a good idea to remove this block?
Removing the stuff directly below is only about self-payments so I am not that concerned about reducing reliability there.
However this block highlighted affects almost all payments and it being there might make attempts more reliable.

Reply:
Yes, you're right, I was too eager there for the outgoing channels. I was a bit mislead, as `active_channels` in `create_routes_for_payment` only selects active sending channels, but does not take into account sending amounts (they would be taken into account with mpp-presplitting). What do you think about what `my_channels` should contain, all our channels or only the sending ones?

Reply:
Either is fine. If there is a benefit in passing only a subset of our channels as `my_channels`, we can do that.
If we do, either we should rename the parameter, or add a docstring/comment to explain what is being passed.